kernel/kernel-impl/src/main/java/org/sakaiproject/site/impl/BaseSiteService.java

```diff
@@ -3575,9 +3578,61 @@ public String merge(String siteId, Element el, String creatorId)
 			}
 		}
 
+		// Add any pages not already in the site - this is needed
+		// when we are importing into an existing site.  When
+		// we are creating a new site (above) all the pages and tools
+		// are added by the "new BaseSite()" with the el parameter.
+		// For sites that were created and then the import happened
+		// the code below makes sure the pages exist and do not get
+		// added twice.
+		try
+		{
+			Site site = getSite(siteId);
+			boolean changed = false;
+			List<SitePage> pages = site.getOrderedPages();
+			Set<String> pageNames = new LinkedHashSet();
+			for (SitePage page : pages) {
+				pageNames.add(page.getTitle());
+			}
```

Comment:
```suggestion
			Set<String> pageNames = pages.stream().map(SitePage::getTitle).collect(Collectors.toCollection(LinkedHashSet::new));
```

Reply:
I am glad you changed this after I debugged it.

Reply:
Lol